ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/app/main.clj

```diff
@@ -1,15 +1,27 @@
 (ns com.sixsq.slipstream.ssclj.app.main
   (:gen-class)
   (:require
-    [com.sixsq.slipstream.ssclj.app.server :refer [start register-shutdown-hook]]
-    [clojure.tools.logging :as log]))
+    [com.sixsq.slipstream.ssclj.app.server :refer [start stop]]))
+
+(def ^:const default-port 8200)
 
 (defn valid-port?
   "If the port number is valid, then returns the port itself;
-   otherwise returns nil."
+   otherwise returns false."
   [port]
-  (if (< 0 port 65536)
-    port))
+  (and (< 0 port 65536) port))
```

Comment:
This will return a _cryptic_ `NPE` or `ClassCastException` if the port is `nil` or a `String`. The issue is that the exception will not be directly related with that function but deeper in the `Numbers` namespace, if we can silently return `false`.

I'd suggest to add a check as a `pre`-condition if we want to be verbose `{:pre [(integer? port)]}` or as the first condition into the existent `and` sform.